[TT-13109]Generate New Swagger and Update Validator for Gateway (#6231)
TykTechnologies/tyk: tyk-api-documentation/README.md

# Tyk API Documentation

The **Tyk API Documentation** is an open api specification with all the endpoints that you can call on the gateway.

## Notes
The current OAS is version 5.3.0 generated from gateway v5.5.0.

The version of Open Api specification used is currently 3.0.3

Currently, we use redocly for linting the OAS. You can install Redocly from (https://github.com/Redocly/redocly-cli)

## How the OAS was generated.

We used the golang library [openapi-go](https://github.com/swaggest/openapi-go) because:

> 1. It supports producing  OASn for version 3. (Golang doesn't to many libraries to produce OAS three)

Reviewer: ```suggestion
1. It supports producing  OAS for version 3. (Golang doesn't have too many libraries that produce OAS 3)
```

Reviewer: @Keithwachira which v3 is it? 3.1?
